core/src/main/scala/io/finch/RequestReaders.scala

```diff
@@ -43,7 +43,7 @@ trait RequestReaders {
     req.cookies.get(cookie)
 
   private[finch] def requestUpload(upload: String)(req: Request): Option[FileUpload] =
-    req.multipart.flatMap(m => m.files.get(upload).flatMap(fs => fs.headOption))
+    Try(req.multipart).getOrElse(None).flatMap(m => m.files.get(upload).flatMap(fs => fs.headOption))
```

Comment:
Oh, we can't use `.toOption` here since it's already `Try[Option[_]]`. Please, ignore this comment.